bin/db/schema_2.sql

```diff
@@ -0,0 +1,95 @@
+DROP TABLE IF EXISTS branches;
+CREATE TABLE branches (
+  id mediumint(9) NOT NULL auto_increment,
+  branch varchar(64) CHARACTER SET utf8mb4 COLLATE utf8mb4_bin DEFAULT '' NOT NULL,
+  PRIMARY KEY (id),
+  UNIQUE branch (branch)
+) ENGINE=InnoDB;
+
+DROP TABLE IF EXISTS commits;
+CREATE TABLE commits (
+  type enum('Change','Add','Remove') DEFAULT NULL,
+  ci_when datetime NOT NULL DEFAULT '1000-01-01 00:00:00',
+  whoid mediumint(9) NOT NULL DEFAULT 0,
+  repositoryid mediumint(9) NOT NULL DEFAULT 0,
+  dirid mediumint(9) NOT NULL DEFAULT 0,
+  fileid mediumint(9) NOT NULL DEFAULT 0,
+  revision varchar(32) CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL DEFAULT '',
+  stickytag varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL DEFAULT '',
+  branchid mediumint(9) NOT NULL DEFAULT 0,
+  addedlines int(11) NOT NULL DEFAULT 0,
+  removedlines int(11) NOT NULL DEFAULT 0,
+  descid mediumint(9) DEFAULT NULL,
+  UNIQUE KEY repositoryid (repositoryid,dirid,fileid,revision),
+  KEY ci_when (ci_when),
+  KEY whoid (whoid),
+  KEY repositoryid_2 (repositoryid),
+  KEY dirid (dirid),
+  KEY fileid (fileid),
+  KEY branchid (branchid),
+  KEY descid (descid)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

Comment:
As all text type fields are explicitly specified to use `CHARACTER SET utf8mb4` , specifying `DEFAULT CHARSET=utf8` here seems to take no effect, however we should specify `DEFAULT CHARSET=utf8mb4` or should not specify `DEFAULT CHARSET`.

Other tables does not specify their `DEFAULT CHARSET`.  Shouldn't we specify it, or should we specify it for all tables?

Reply:
I went with "specify the character set and collation as defaults on all tables instead of per-column designations".